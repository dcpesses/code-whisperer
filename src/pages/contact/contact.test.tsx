import {render, screen} from '@testing-library/react';
import Contact from './index';
import {userEvent} from '@testing-library/user-event';
// import * as ContactForm from '@/components/contact-form';

interface LinkProps {
  className: string,
  to: string,
  children: React.ReactNode
}

global.fetch = vi.fn();

vi.mock('react-router-dom', () => {
  const reactRouterDom = vi.importActual('react-router-dom');
  return {
    ...reactRouterDom,
    Link: ({className, to, children}: LinkProps) => (
      <div className={className} data-to={to}>
        {children}
      </div>
    )
  };
});

describe('Contact', () => {
  test('Should render Contact page using default props', async() => {
    render(<Contact />);

    expect(screen.getByText('Contact Us')).toBeTruthy();

    const btnSubmit = screen.getByRole('button', { name: 'Submit' });

    await userEvent.click(btnSubmit);

    expect(screen.getByTestId('contact')).toBeDefined();
    expect(screen.getByTestId('contact')).toMatchSnapshot();
  });
});
