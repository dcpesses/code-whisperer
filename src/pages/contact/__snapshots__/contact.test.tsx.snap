// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Contact > Should render Contact page using default props 1`] = `
<div
  class="contact container"
  data-testid="contact"
>
  <main
    class="px-3"
  >
    <h1
      class="py-4"
    >
      Contact Us
    </h1>
    <form
      accept-charset="UTF-8"
      action="/thanks"
      class="was-validated"
      data-testid="contact-form"
      enctype="multipart/form-data"
      id="contact-form"
      method="GET"
      novalidate=""
    >
      <div>
        <label
          class="form-label"
          for="contact.Full_Name"
        >
          Full Name *
        </label>
        <input
          class="form-control"
          id="contact.Full_Name"
          name="Full Name"
          placeholder="First name"
          required=""
          type="text"
        />
        <div
          class="invalid-feedback"
        >
          Please enter your full name
        </div>
      </div>
      <div>
        <label
          class="form-label"
          for="contact.Email_Address"
        >
          Email Address *
        </label>
        <input
          class="form-control"
          id="contact.Email_Address"
          name="Email Address"
          required=""
          type="email"
        />
        <div
          class="invalid-feedback"
        >
          Please enter a valid email address
        </div>
      </div>
      <div>
        <label
          class="form-label"
          for="contact.Comments_Questions"
        >
          Comments/Questions *
        </label>
        <textarea
          class="form-control"
          id="contact.Comments_Questions"
          name="Comments/Questions"
          required=""
        />
        <div
          class="invalid-feedback"
        >
          Please enter your feedback.
        </div>
      </div>
      <div
        style="opacity: 0; position: absolute; top: 0px; left: -5000px; height: 0px; width: 0px;"
      >
        <label
          for="subscribe_a7fdfc1ea41e_48062"
        />
        <input
          autocomplete="off"
          id="email_subscribe_a7fdfc1ea41e_48062"
          name="subscribe_a7fdfc1ea41e_48062"
          placeholder="Your email here"
          tabindex="-1"
          type="email"
          value=""
        />
      </div>
      <input
        autocomplete="off"
        id="g-recaptcha-response"
        name="g-recaptcha-response"
        type="hidden"
        value=""
      />
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </form>
    <div
      class="col-6 mx-auto"
    >
      <div
        class="btn btn-outline-primary btn-lg mt-5"
        data-to="/"
      >
        Go Home
      </div>
    </div>
  </main>
</div>
`;
